__tests__/notification.pane/notification.action.test.js

```diff
@@ -0,0 +1,185 @@
+				const result = store.getActions();
+				expect(result).toEqual(expectation);
+			});
+	});
+	it('should get all notifications more than 10', async () => {
+		localStorage.setItem('token', token);
+		const expectation = [
+			{
+				type: 'CHANGE_NOTIFICATION_PANE_HEIGHT',
+				notificationPaneHeight: 650,
+				viewButton: false,
+				limitNumber: 20,
+				buttonTitle: 'less',
+			},
+		];
+		const store = mockStore({});
+		await store
+			.dispatch(notificationPaneHeigthHandle(20, [true]))
+			.then(async () => {
+				const result = store.getActions();
+				expect(result).toEqual(expectation);
+			});
+	});
+
+	it('should mark notifications as read', async () => {
+		localStorage.setItem('token', token);
+		moxios.wait(() => {
+			const request = moxios.requests.mostRecent();
+			request.respondWith({
+				status: 200,
+				message: 'success',
+				response: {
+					data: {
+						data: ['New message', 'New trips'],
+					},
+				},
+			});
+		});
+
+		const expectation = [
+			{
+				markNotificationAsReadMessage: { data: ['New message', 'New trips'] },
+				type: 'MARK_NOTIFICATIONAS_READ',
+			},
+		];
+
+		const store = mockStore({});
+		await store.dispatch(markNotificationAsRead(2)).then(async () => {
+			const result = store.getActions();
+			expect(result).toEqual(expectation);
+		});
+	});
+
+	it('should get all notifications after mark all notifications as read', async () => {
+		localStorage.setItem('token', token);
+		moxios.wait(() => {
+			const request = moxios.requests.mostRecent();
+			request.respondWith({
+				status: 200,
+				message: 'success',
+				response: {
+					data: {
+						data: ['New message', 'New trips'],
+					},
+				},
+			});
+		});
+		const store = mockStore({});
+		await store.dispatch(fetchUserNotification()).then(async () => {
+			const result = store.getActions();
+			expect(result[0].type).toEqual('GET_NOTIFICATIONS');
+		});
+	});
+	it('should get notification via socket io', async () => {
+		const initialState = {};
+		const store = mockStore(initialState);
+		store.dispatch(getUserNotificationSocket({ name: 'shema' }));
+		const actions = store.getActions();
+		expect(actions[0].type).toEqual('GET_SOCKET_NOTIFICATIONS');
+	});
+	it('should cencing notifications from server via socket', () => {
+		const initialState = {};
+		const store = mockStore(initialState);
+		store.dispatch(socketManagement(1));
+		serverSocket.emit('approve-or-reject-trip_request_event', () => ({
+			title: 'new notification',
+			message: ' trip approved',
+		}));
+		serverSocket.emit('edit-trip-request', () => ({
+			title: 'new notification',
+			message: ' trip edited',
+		}));
+		serverSocket.emit('trip_request_comment_event', () => ({
+			title: 'new notification',
+			message: ' comment',
+		}));
+		serverSocket.emit('trip_request_event', () => ({
+			title: 'new notification',
+			message: ' trip created',
+		}));
+		const actions = store.getActions();
+		expect(actions.length).toEqual(4);
+	});
+});
```

Comment:
@shemaeric1234 why are the following codes out test suits?

Reply:
I did this because the response I'm getting includes the current date, so I'm not able to mock that date.

Reply:
I have done